src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -144,21 +144,25 @@ public void Prepare(CancellationToken token)
                 }
 
                 var info = keyValue.Value;
-                Debug.Assert(!string.IsNullOrWhiteSpace(info.FilePath), "Expected to be set during registration time");
 
-                sourceControlledFileSystem.QueueFileWrite(info.FilePath, () =>
+                foreach (var solutionItem in info.SolutionItems)
                 {
-                    var ruleSetDirectoryPath = Path.GetDirectoryName(info.FilePath);
+                    Debug.Assert(!string.IsNullOrWhiteSpace(solutionItem), "Expected to be set during registration time");
 
-                    fileSystem.Directory.CreateDirectory(ruleSetDirectoryPath); // will no-op if exists
+                    sourceControlledFileSystem.QueueFileWrite(solutionItem, () =>
+                    {
+                        var ruleSetDirectoryPath = Path.GetDirectoryName(solutionItem);
 
-                    // Create or overwrite existing rule set
-                    info.Save();
+                        fileSystem.Directory.CreateDirectory(ruleSetDirectoryPath); // will no-op if exists
 
-                    return true;
-                });
+                        info.Save();
```

Comment:
This is really odd api :/
We will re-save all the files X number of files.

Reply:
Can the `sourceControlledFileSystem` be changed easily to support `QueueFileWrites(IEnumerable<string>)` ?